[TECH] Renommer badge-partner-competence en skillset (PIX-2708).

celineung opened the pull request:
## :jack_o_lantern: Problème
Suite à la PR #3112, on parle désormais de `SkillSet` pour les lots d'acquis.
Historiquement appelé `BadgePartnerCompetence`, cette PR renomme `BadgePartnerCompetence`en `SkillSet`.

## :bat: Solution
Renommer `BadgePartnerCompetence`en `SkillSet`.

## :spider_web: Remarques
On a parfois des `BadgePartnerCompetence` et parfois des `PartnerCompetence`.
Or il s'agit de la même chose. Ces deux éléments ont été renommés en `SkillSet`.

Puisque le front met plus de temps à se déployer que l'api, le dernier commit de cette PR permet d'envoyer à la fois `badgePartnerCompetence` et `skillSet`. **Ce dernier commit sera à revert une fois la PR déployée en prod**

## :ghost: Pour tester
Se rendre sur la RA.
Gagner des badges sur les campagnes.

Constater qu'en fin de campagne:
- sur la campagne AZERTY123: les noms des compétences Pix sont affichées
- sur la campagne CLEA123: les noms des compétences SkillSet sont affichées

(J'ai créé la campagne CLEA123 pour les tests de cette RA. Actuellement c'est la campagne QWERTY789 qui fait gagner le badge `PIX_EMPLOI_CLEA_V2`, mais il faut se connecter via pôle emploi pour la passer)

Commit:
feat(api)(db-migration): rename badge-partner-competences to skillsets

Commit:
feat(api): rename partnerCompetenceIds to skillIds in models/BagdeCriterion

Commit:
feat(api)(seeds): rename badge-partner-competence to skill-set

Commit:
feat(api): rename BadgePartnerCompetence to SkillSet in databaseBuilder

Commit:
feat(api): rename partnerCompetence in services

Commit:
feat(api): rename badgePartnerCompetence in repositories

Commit:
feat(api): rename badgePartnerCompetence in models

Commit:
feat(api): rename partnerCompetence in read-models

Commit:
feat(api): rename partnerCompetence in serializers

pix-service commented:
I'm deploying this PR to these urls:

- App (.fr): https://app-pr3613.review.pix.fr
- App (.org): https://app-pr3613.review.pix.org
- Orga: https://orga-pr3613.review.pix.fr
- Certif: https://certif-pr3613.review.pix.fr
- Admin: https://admin-pr3613.review.pix.fr
- API: https://api-pr3613.review.pix.fr/api/

 Please check it out!

yannbertrand commented:
API 🆗 pour moi ✅

alexandrecoin commented:
Ok pour la partie API ✌️

Commit:
feat(admin): rename badgePartnerCompetence to skillSet

Commit:
feat(mon-pix): rename badge-partner-competence to skill-set

VincentHardouin commented:
⚠️ il faut laisser une période de transition où l'on peut appeler l'API avec les anciens noms, car l'API et le front ne se déploie pas en même temps. Cela a pour conséquence de laisser un état où l'API n'accepte plus l'ancien nom car elle se déploie plus vite mais le front lui appel toujours avec l'ancien nom.

Commit:
fix(api): keep using both badgePartnerCompetence and skillSet in serialisers

This commit is to be revert after deployment in production:
Because the deployment of api happens faster than mon-pix, there could be a few minutes where the old version of mon-pix queries the newly released api.

MelanieMEB commented on api/lib/infrastructure/serializers/jsonapi/badge-with-learning-content-serializer.js:
```diff
@@ -19,17 +20,37 @@ module.exports = {
         include: true,
         ref: 'id',
-        attributes: ['threshold', 'scope', 'partnerCompetences'],
+        attributes: ['threshold', 'scope', 'skillSets', 'partnerCompetences'],
```
```suggestion
        attributes: ['threshold', 'scope', 'skillSets'],
```
On en a plus besoin normalement

celineung replied:
@MelanieMEB on garde `partnerCompetences` pour s'assurer que la version du front qui utilise encore `partnerCompetences` fonctionne (le front met plus de temps à se déployer que l'api)

MelanieMEB commented on api/lib/infrastructure/serializers/jsonapi/badge-with-learning-content-serializer.js:
```diff
@@ -19,17 +20,37 @@ module.exports = {
+          include: false,
+          ref: 'id',
+        },
         partnerCompetences: {
           include: false,
           ref: 'id',
         },
       },
```
```suggestion
```

MelanieMEB commented on api/lib/infrastructure/serializers/jsonapi/badge-with-learning-content-serializer.js:
```diff
@@ -51,19 +72,21 @@ module.exports = {
             skill.tube = { ...record.tubes.find(({ id }) => id === skill.tubeId) };
           });
         });
+        badge.badgePartnerCompetences = badge.skillSets;
```
Question : Pourquoi on en a encore besoin sur cette route ?

MelanieMEB reviewed (changes requested):
Test fonctionnel : ca a l'air ok.

VincentHardouin commented:
J'ai testé en faisant un revert des commits des fronts c'est OK.

MelanieMEB reviewed (approved):
Ok pour moi !
